restaurant-app/index.js

```diff
@@ -0,0 +1,343 @@
+        <h3>${name}</h3>
+        <p>${cost}</p>
+		    <p id="category">${category}</p>
+        </li>
+        `;
+      menuId.insertAdjacentHTML("beforeend", item);
+    }
+  }
+
+
+  displayTables();
+  displayMenu();
+
+  // function to search tables.
+  function searchTable() {
+    let searchKey = document.getElementById("table-search").value;
+    
+    searchKey = searchKey.toLowerCase();
+    tableId.innerHTML = "";
+    searchKey = searchKey.split(" ").join("");
+    for( i = 0; i < NO_OF_TABLES; i++)
+    {
+        if( tables[i].table_name.includes(searchKey) )
+        {
+            let id = i+1;
+            let cost = tables[i].cost; 
+            let items = tables[i].items;
+            let tableSearch = `<li onclick="openModal('table${id}')" ondrop="drop(event,'table${id}')" ondragover="allowDrop(event)">
+            <h2>Table ${id}</h2>
+            <p>Rs ${cost} | Total items: ${items}</p>
+            </li>`;
+        
+            tableId.insertAdjacentHTML("beforeend", tableSearch);
+        }
+    }
+  }
+
+  //function to search items in menu
+  function searchMenu() {
+    let searchKey = document.getElementById("menu-search").value;
+    searchKey = searchKey.toLowerCase();
+  
+    let i = 1;
+  
+    if (searchKey == "") {
+      return;
+    }
+  
+    menuId.innerHTML = "";
+  
+    while (menu["item" + i] != null) {
+      let { name, cost, category } = menu["item" + i];
+      let itemName = name.toLowerCase();
+      let itemCategory = category.toLowerCase();
+  
+      if (itemName.includes(searchKey) || itemCategory.includes(searchKey)) {
+        let item = `<li id="item${i}" draggable="true" ondragstart="drag(event)">
+        <h2>${name}</h2>
+		<p id="category">${category}</p>
+        <p>${cost}</p>
+        </li>`;
+        menuId.insertAdjacentHTML("beforeend", item);
+      }
+      i++;
+    }
+  }
+
+  //Drop start
+function drag(e) {
+    e.dataTransfer.setData("id", e.target.id);
+  }
+  
+  //DropOver
+  //By default dropping is not allowed but we disable it
+  function allowDrop(e) {
+    e.preventDefault();
+  }
+  
+  //After dropping the item on table
+  function drop(e, tableName) {
+    addItemToTable(tableName, e.dataTransfer.getData("id"));
+  }
+  
+  //For adding item to the table after dropping
+  function addItemToTable(tableName, itemId) {
+    let item = menu[itemId];
+    let tableNumber = tableName[5] - 1; 
+    if (tables[tableNumber]["orders"][itemId] == undefined) {
+      tables[tableNumber]["orders"][itemId] = 1;
+    } else {
+      tables[tableNumber]["orders"][itemId] += 1;
+    }
+  
+    //Displaying cost of items
+    tables[tableNumber].cost += item.cost;
+    tables[tableNumber]["items"] += 1;
+    displayTables();
+  }
+
+  //Modal close
+function closeModal() {
+  modal.style.display = "none";
+}
+  
+// Adding content to Modal
+function openModal(tableNo) {
+  modal.style.display = "block";
+  let tableNumber = tableNo[5];
```

Comment:
What is tableNo[5]

Reply:
I took tableNo(which is a type String) as parameter 
Here tableNo can be 'table1' or 'table2' or 'table3' (These are id's of the Tables), to access the table number which is at index no. - 5 . I used tableNo[5].
tableNo[5] = '1' or '2' or '3'

Reply:
Why it is only table 5? It should be generic.